packages/core/src/components/graphs/gauge.ts

```diff
@@ -210,7 +210,14 @@ export class Gauge extends Component {
 			.merge(valueNumber as any)
 			.style('font-size', `${fontSize}px`)
 			.attr('text-anchor', 'middle')
-			.text((d: any) => localeNumberFormatter(Number(numberFormatter(d)), localeCode))
+			.text((d: any) => {
+				const value = Number(d.toFixed(2)) % 1 !== 0 ? d.toFixed(2) : d.toFixed()
+				if (numberFormatter) {
+					return numberFormatter(value)
+				} else {
+					return localeNumberFormatter(Number(value), localeCode)
```

Comment:
is there ever a chance of `localeNumberFormatter` being null or undefined? we should have a fallback probably